src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/AwsSdkPipelineWrapper.cs

```diff
@@ -14,12 +18,56 @@ public class AwsSdkPipelineWrapper : IWrapper
 
         private const string WrapperName = "AwsSdkPipelineWrapper";
         private static ConcurrentHashSet<string> _unsupportedRequestTypes = new();
+        private static bool _reportBadAccountId = true;
+        private static bool _reportBadArnBuilder = false;
 
         public CanWrapResponse CanWrap(InstrumentedMethodInfo methodInfo)
         {
             return new CanWrapResponse(WrapperName.Equals(methodInfo.RequestedWrapperName));
         }
 
+        private ArnBuilder CreateArnBuilder(IAgent agent, dynamic requestContext)
+        {
+            string partition = null;
+            string systemName = null;
+            string accountId = null;
+            try
+            {
+                accountId = GetAccountId(agent);
+                var clientconfig = requestContext.ClientConfig;
+                var regionEndpoint = clientconfig.RegionEndpoint;
+                systemName = regionEndpoint.SystemName;
+                partition = regionEndpoint.PartitionName;
+            }
+            catch (Exception e)
+            {
+                if (_reportBadArnBuilder)
+                {
+                    agent.Logger.Debug(e, $"AwsSdkPipelineWrapper: Unable to get required ARN components from requestContext.");
+                    _reportBadArnBuilder = false;
+                }
+            }
+
+            return new ArnBuilder(partition, systemName, accountId); ;
+        }
+
+        private string GetAccountId(IAgent agent)
+        {
+            string accountId = agent.Configuration.AwsAccountId;
+            if (accountId != null)
```

Comment:
it would be more performant to check `_reportBadAccountId` here - if it's `false`, none of the rest of the validation needs to be done.

Reply:
If that's the case, should the validation be done in the DefaultConfiguration class instead?

Reply:
I had planned on revisiting this after the ID decoding was done. There were a few things I wanted to consider:
* Having the validation in a central place
* We may want to fall back to the other ID method if one is invalid
* The environment variable for the ID may be set after we've read the config
* Depending on how we found the ID, it may be bad at first but good later